hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -120,6 +121,7 @@
   protected transient AsyncArchiveService asyncArchiveService;
   protected final TransactionManager txnManager;
   protected Option<Pair<HoodieInstant, Map<String, String>>> lastCompletedTxnAndMetadata = Option.empty();
+  protected List<HoodieInstant> pendingReplaceRequestedInstants = new ArrayList<>();
```

Comment:
Can the issue also occur for ingestion commits as well, not just for replacecommits?
For example: 
c1.commit
c2.inflight
c3.complete
c4.requested(This is based out of c3 commit)
Here lastCompletedTxnOwnerInstant will still be c3 and if c2 has a conflict with c4 it might cause an issue.

While fetching from the getCandidateInstants method especially with findInstantsAfter, should we consider transition times for ordering the instants?

Reply:
good question， Let me dig deeper.

Reply:
+1.  This can happen for multi-writer and echoes my comment above.  The c2 is from backfill and takes a long time to finish, that can cause the same issue as the replacecommit case.